pkg/controllers/jobset_controller.go

```diff
@@ -129,6 +130,55 @@ func (r *JobSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		return ctrl.Result{}, nil
 	}
 
+	// If JobSet is suspended, ensure the suspend condition is set and suspend all active child jobs.
+	jobsetSuspended := js.Spec.Suspend != nil && *js.Spec.Suspend
+	if jobsetSuspended {
+		if err := r.ensureCondition(ctx, &js, corev1.EventTypeNormal, metav1.Condition{
+			Type:               string(jobset.JobSetSuspended),
+			Status:             metav1.ConditionStatus(corev1.ConditionTrue),
+			LastTransitionTime: metav1.Now(),
+			Reason:             "SuspendedJobs",
+			Message:            "jobset is suspended",
+		}); err != nil {
+			log.Error(err, "updating jobset status")
+			return ctrl.Result{}, nil
+		}
```

Comment:
pls move this to after the loop below, we should suspend the jobs first and then update the status.

Reply:
looks like this was done

Reply:
Thanks!